src/main/java/org/folio/circulation/domain/RequestRepresentation.java

```diff
@@ -53,74 +40,61 @@ private static void addStoredItemProperties(JsonObject request, Item item) {
     request.put("item", itemSummary);
   }
 
-  private static void addStoredRequesterProperties(
-    JsonObject requestWithAdditionalInformation,
-    User requester) {
+  private static void addStoredRequesterProperties(JsonObject request, User requester) {
 
-    if(requester == null) {
+    if (requester == null) {
+      log.info(UNABLE_TO_ADD_REQUESTER_PROPERTIES_MSG, request.getString("id"));
```

Comment:
I'm curious.

I personally prefer the log message to be inline, as I find it easier to comprehend what the likely output is, without having to navigate back and forth.

@Anatolii-Starkov Was the motivation for extracting constants reuse, shortening of lines, or something else?

If the logging is unwieldy, I think I'd prefer that the log itself be extracted to a self contained method and called.

Given that no one else has remarked upon this change, I imagine I might be an outlier. @mreno-EBSCO @mtraneis @jsavell What do you think, any strong preference?

(and I think FOLIO needs better guidance about what to log and in what structure)

Reply:
@marcjohnson-kint I agree. The static log messages tend clutter up other statics (not in this case, since the log messages are the only statics, besides the logger) and need to be created on class load. I'm not sure if these messages are intended to be logged in common cases or exceptional cases - many are logged when something is null, which I hope is somewhat uncommon.

I think if we are going to externalize messages they are better stored in a java resource bundle (like a properties file) for easy translation later on as well as standards based loading (well, java standards 😄).

Sonar will flag multiple uses of the same string for extraction into a constant. I didn't look at all the uses, so maybe some of these constants were to eliminate Sonar code smells and for consistency all were moved to constants.

I am pretty sure Eclipse will show the values of the constants if hovered over, but again, having the string inline is a lot easier to quickly grok.

Also, some of these messages seem more like debug level messages rather than info messages. Not a big deal, but I'm not sure if all these messages help with diagnosis or just clutter up the log.

Reply:
@mreno-EBSCO 
> Sonar will flag multiple uses of the same string for extraction into a constant. I didn't look at all the uses, so maybe some of these constants were to eliminate Sonar code smells and for consistency all were moved to constants.

Yeah, I've been increasingly trying to take the alternative approach of extracting a method for the repeated use. Though this will not be appropriate in all cases.

> Also, some of these messages seem more like debug level messages rather than info messages. Not a big deal, but I'm not sure if all these messages help with diagnosis or just clutter up the log.

We do need greater guidance on structure and level of log messages, and then a good clean up of the existing log messages.

Reply:
@marcjohnson-kint My motivation is, I think logging is not business related logic and it should take place in a method (on my monitor) as less as possible. It's sort of a noise. Short variable name that describes a message I think much better than 2 or 3 lines of useless code and of course I vote with both hands when messages are in properties file but it will make me refactor whole project(I don't mind but it should be under another ticket).

According to cluttering up statics stuff. I don't see any problem, How much time do we spend reading statics stuff?

Reply:
@Anatolii-Starkov I reckon we likely just have different preferences.

> Short variable name that describes a message I think much better than 2 or 3 lines of useless code

If we think the logging is useless, then I think it should be removed, rather than restructured.

If we think the logging is valuable for diagnosis of production issues. Then I agree, it should distract from other responsibilities as much as possible.

> I think logging is not business related logic and it should take place in a method (on my monitor) as less as possible.

Are you saying that you find logging extracted to a method more distracting than the inline form with a static context?

To me, hiding the logging in a method that clearly states that logging is it's only responsibility, is a way to reduce the distraction from the business logic that you refer to above.

If the goal is to reduce distracting, I think I personally find that less distracting than either inline form. It seems you believe the shortened inline form is less distracting than a method, or have I misunderstood?

Reply:
> If the goal is to reduce distracting, I think I personally find that less distracting than either inline form. It seems you believe the shortened inline form is less distracting than a method, or have I misunderstood?
Could you give us an example to how it looks like?

Reply:
Sure, something like (I haven't checked the syntax of this):

```suggestion
  if (requester == null) {
    logUnableToAddRequesterProperties(request.getString("id"));
    return;
  }
  
  private void logUnableToAddRequesterProperties(String requestId) {
    log.info(UNABLE_TO_ADD_REQUESTER_PROPERTIES_MSG, requestId);
  }
```

I likely wouldn't have the constant if the logging is extracted, but this was the simplest example I could construct.

Reply:
@Anatolii-Starkov @mreno-EBSCO @mtraneis @jsavell What do people think about the different logging approaches?

Reply:
I think constant-approach is more neat, method-approach brings much more lines of code. Anyway we need to make a decision.

Reply:
@mreno-EBSCO @mtraneis Any strong opinions either way?

Reply:
I can't say I've used either approach for log messages.

I don't like constants because they aren't part of the business logic of the module and we shouldn't be making decisions based on a log message, so I view them as clutter. They are also loaded for the life of the VM even if they aren't used.

I think creating a method is fine if you need to construct the message in some way or the message is massive. But it seems a bit overkill for the single log line case.

In both cases there could be hundreds of log messages in a class with a lot of business logic or where trace level logging is needed for diagnostics (like over the wire stuff). Also, I find it easier to read the message to understand what we are trying to convey then it is to read a summarized constant or method name.

I've always thought the best approach, and the one I typically use, is to inline the message in the log line. If the messages need i18n, then I use a `ResourceBundle`.

If we are concerned with repeated message use, why not break out the check into a method like `isRequesterNull` and to the null check, if `true` log and return `true` else `false`? Then use this method for each of these checks. At least this method conveys some business logic.

Sorry about the rambling comment. 😴

Reply:
I don't have a preference on these approaches, either logging with inline messages or creating some sort of abstractions around it is fine. If we find that the logging code clutters up the real domain code, then we need to abstract it out. 

I like the approach of putting the logging code in a method, but if that method is not reusable in many places then it doesn't achieve much.  Using constants is fine, too, but not so good for readability and like @mreno-EBSCO said these constants aren't part of domain logic.  Also, if we don't reuse them then they don't serve much purpose.  Alternatively we could put the log messages or constants in a different "logging" class, but then we're leaking the domain logic outside of this class. 

Looking back at the class as a whole to review the motivation of abstracting out constants, I really don't think the constants help cleaning up the code here.  We're just moving the hard-coded string messages around.

Reply:
I removed constant log statements and left them as they were before my refactoring.